components/nostr-auth.js

```diff
@@ -74,83 +89,186 @@ export function NostrAuth ({ text, callbackUrl, multiAuth }) {
+        tags: [['challenge', k1]],
+        content: 'Stacker News Authentication'
+      }, { signer })
+
+      await signIn('nostr', {
+        event: JSON.stringify(signedEvent),
+        callbackUrl,
+        multiAuth
+      })
+    } catch (e) {
+      setError(e)
+    }
+  }, [])
 
   return (
     <>
-      {hasExtension === false && <NostrExplainer text={text} />}
-      {extensionError && <ExtensionError {...extensionError} />}
-      {hasExtension && !extensionError &&
-        <>
-          <h4 className='fw-bold text-success pb-1'>nostr extension found</h4>
-          <h6 className='text-muted pb-4'>authorize event signature in extension</h6>
-        </>}
+      {status.error && <NostrError message={status.msg} />}
+      {status.loading
+        ? (<Moon className='spin fill-grey' width='50' height='50' />)
+        : (
+          <>
+            <Row className='w-100 g-1'>
+              <Form
+                initial={{ token: '' }}
+                onSubmit={values => {
+                  if (!values.token) {
+                    setError(new Error('Token or NIP-05 address is required'))
+                  } else {
+                    auth(values.token)
+                  }
+                }}
+              >
+                <Input
+                  label='Connect with token or NIP-05 address'
+                  name='token'
+                  placeholder='bunker://...  or NIP-05 address'
+                  required
+                  autoFocus
+                />
+                <div className='mt-2'>
+
+                  <SubmitButton className='w-100' variant='primary'>
+                    {text || 'Login'} with token or NIP-05
+                  </SubmitButton>
+
+                </div>
+              </Form>
+              <div className='text-center text-muted fw-bold'>or</div>
+              <Button
+                variant='nostr'
+                className='w-100'
+                type='submit'
+                onClick={async () => {
+                  try {
+                    await auth()
+                  } catch (e) {
+                    setError(e)
+                  }
+                }}
+              >
+                {text || 'Login'} with extension
+              </Button>
+            </Row>
+            <Row className='w-100 mt-4 text-muted small'>
+              <AccordianItem
+                header='Which NIP-46 signers can I use?'
+                body={
+                  <>
+                    <Row>
+                      <Col>
+                        <ul>
+                          <li>
+                            <a href='https://nsec.app/'>Nsec.app</a><br />
+                            available for: chrome, firefox, and safari
+                          </li>
+                          <li>
+                            <a href='https://app.nsecbunker.com/'>nsecBunker</a><br />
+                            available as: SaaS or self-hosted
```

Comment:
the original PR had more signers in this list, but i've found they have issues or don't seem to be very well maintained, so i've decided to keep only these two